webrender/src/device/gfx/device.rs

```diff
@@ -1358,17 +1367,17 @@ impl<B: hal::Backend> Device<B> {
         &mut self,
         descriptor_group: DescriptorGroup,
         per_draw_bindings: PerDrawBindings,
-    ) -> DescriptorSet<B> {
+    ) {
         let mut bound_textures = self.bound_textures;
         bound_textures[0] = per_draw_bindings.0[0];
         bound_textures[1] = per_draw_bindings.0[1];
         bound_textures[2] = per_draw_bindings.0[2];
         let mut bound_sampler = self.bound_sampler;
-        bound_sampler[0] = per_draw_bindings.1[0];
-        bound_sampler[1] = per_draw_bindings.1[1];
-        bound_sampler[2] = per_draw_bindings.1[2];
+        bound_sampler[0] = per_draw_bindings.1;
```

Comment:
can we have that to be just `bind_sampler = ` instead of having a fixed size array of samplers still?

Reply:
`bound_sampler` is just for tracking the kind of sampler we will use for a particular texture slot, those are not the actual samplers we use for binding descriptors (Those are located here https://github.com/szeged/webrender/blob/master/webrender/src/device/gfx/device.rs#L253-L254). Since we have multiple textures and they could use a different combination of those samplers for a draw, I think it would be wiser to keep around that `bound_sampler` .

Reply:
Can they really use different samplers? my understanding is that all the textures here should use the same sampler in the shader

Reply:
The three texture here uses the same sampler and we are doing that now. But it's not true for all the textures that's why we need to keep that array of sampler solution.

Reply:
My feeling is that this should be the case for any 3 textures that we'd want to bind here. Do you know what case doesn't fit this?

Reply:
There is no such case, I just wanted to highlight that the array we use for tracing samplers used for all the textures not just for these three.